app/models/dataset_record/darwin_core/occurrence.rb

```diff
@@ -266,21 +266,23 @@ def parse_people(field_name)
     end
   end
 
+  # Parse the
+  # @param [String] field_name
   def parse_iso_date(field_name)
     value = get_field_value(field_name)
 
-    if value.nil?
-      return nil
-    end
+    return nil if value.nil?
 
     result = Utilities::Dates.parse_iso_date_str(get_field_value(field_name))
     if result.nil?
       raise DarwinCore::InvalidData.new(
-        { "#{field_name}":
+        {
+          "#{field_name}":
             ["Invalid date. Please make sure it conforms to ISO 8601 date format (yyyy-mm-ddThh:mm:ss). If expressing interval separate result with '/'. Examples: 1972-05; 1983-10-25; 2020-09-22T15:30; 2020-11-30/2020-12-04"]
         }
       )
```

Comment:
You may move `if result.nil?` here to do the same thing you did with `return nil`

Reply:
What do you mean?

Reply:
Replace
```ruby
    if result.nil?
      raise DarwinCore::InvalidData.new(
        {
          "#{field_name}":
            ["Invalid date. Please make sure it conforms to ISO 8601 date format (yyyy-mm-ddThh:mm:ss). If expressing interval separate result with '/'. Examples: 1972-05; 1983-10-25; 2020-09-22T15:30; 2020-11-30/2020-12-04"]
        }
      )
    end
```
With:
```ruby
    raise DarwinCore::InvalidData.new(
      {
        "#{field_name}":
          ["Invalid date. Please make sure it conforms to ISO 8601 date format (yyyy-mm-ddThh:mm:ss). If expressing interval separate result with '/'. Examples: 1972-05; 1983-10-25; 2020-09-22T15:30; 2020-11-30/2020-12-04"]
      }
    )  if result.nil?
```